### PR TITLE
feat: add near wallet

### LINKED ISSUE #2602: [Bounty] Integrate NEAR wallet into Mask
## Catalogue

- Platform
  - [x] Gitcoin
  - [ ] Hackerlink
  - [ ] Mask
- Experience
  - [ ] Beginner
  - [ ] Intermediate
  - [x] Advanced
- Category
  - [ ] Mask Core Feature
  - [ ] Mask Plugin
  - [ ] Security/Cryptography Feature
  - [ ] Documentation
  - [ ] Design
  - [x] Other
    - [x] Cryptocurrency Wallet
- Time Commitment
  - [x] 4 weeks
- Bounty Size
  - [x] 5000 USD

## Part 1: Problem Description
NEAR exists to accelerate the world’s transition to open technologies by growing and enabling a community of developers and creators. NEAR is a decentralized application platform that secures high value assets like money and identity with the performance necessary to make them useful for everyday people, putting the power of Open Finance and the Open Web in their hands. NEAR’s unique account model allows developers to build secure apps that consumers can actually use similarly to today’s web apps, something which requires multiple second-layer add-ons on other blockchains.

As the first step of our cooperation, we, Mask Network, would like to integrate NEAR as one of our default supported blockchain system in our internal Mask Wallet, so that further on-chain interactions would happen directly within Mask Network without leaving the platform. We believe this would further help us attract more users from different projects, aiming at different goals to maximize users experiences in the broad blockchain world.

We are now renovating our internal wallet and targeting at finishing that by the end of April. If you are interested in this bounty, please make sure that you contact us first to see how our structure looks like and proceed to the development afterwards.

## Part 2: Resources
https://wallet.near.org/
https://github.com/DimensionDev/Maskbook/tree/master/packages/maskbook/src/plugins/Wallet

## Part 3: Contact and Support
@yisiliu @septs @Jack-Works 
[Mask Network Discord Server](https://discord.gg/4Xccj7a75Q)

### PR DESCRIPTION
**Pull request by developerfred**
closes #2602

**WIP...** 

**resources**
- [Near swap](https://github.com/kamescg/near-swap)
- [Near Wallet](https://github.com/near/near-wallet)
- [Near Bridge](https://github.com/near/rainbow-bridge-client)
- [Near RPC api](https://github.com/near/near-api-js)
- [Near seed](https://github.com/near/near-seed-phrase)
- [Near web3 provider](https://github.com/near/near-web3-provider)

### CODE REVIEW

**Comment by yisiliu** [comment]
One thing before heading into the tunnel: Have you considered if we are going to support email or phone recovery in our wallet as is provided by NEAR wallet?

File: packages/maskbook/package.json [hunk 1]
```diff
@@ -171,5 +171,8 @@
   },
   "peerDependencies": {
     "jss": "*"
+  },
+  "volta": {
```
Comment (Jack-Works) [inline packages/maskbook/package.json:175]: What is this?

File: packages/maskbook/src/resources/NearIcon.tsx [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import { SvgIcon, SvgIconProps } from '@material-ui/core'
+
+export function NearIcon(props: SvgIconProps) {
```
Comment (Jack-Works) [inline packages/maskbook/src/resources/NearIcon.tsx:3]: Hi I suggest to add new icons at `packages\icons`

Comment (septs) [reply packages/maskbook/src/resources/NearIcon.tsx:3]: plz use https://aka.pw/svgomg reduce size

Comment (developerfred) [reply packages/maskbook/src/resources/NearIcon.tsx:3]: Thanks @septs and @Jack-Works

File: packages/maskbook/package.json (hunk 1 above)
Comment (developerfred) [reply packages/maskbook/package.json:175]: @Jack-Works this is my node version manager, it shouldn't be there. lol

**Comment by developerfred** [comment]
> One thing before heading into the tunnel: Have you considered if we are going to support email or phone recovery in our wallet as is provided by NEAR wallet?

I thought about supporting it, but I still don't know what the layout would be like.

**Pull request closed without merging**